src/operator/tensor/broadcast_reduce_op.h

```diff
@@ -168,15 +168,24 @@ struct BroadcastLikeParam : public dmlc::Parameter<BroadcastLikeParam> {
   }
 };
 
-inline int CheckAxis(int axis, int ndim) {
-  CHECK(axis < ndim && axis >= -ndim)
-    << "axis " << axis << " exceeds the input dimension of " << ndim;
-  return (axis + ndim)%ndim;
+inline int CheckAxis(const int axis, const int ndim) {
+  if (ndim == 0) {
+    CHECK(axis == 0 || axis == -1) << "axis " << axis << " is out of bounds for array of"
+                                                         " dimension 1";
```

Comment:
Should be dimension 0?

Reply:
I suggest putting these common checking utilities to another place so we can reuse the code. May be in a later PR.

Reply:
I agree, however seems like this is the only place that uses this. I think maybe `include/mxnet/operator_utils.h` is a good place?

Reply:
I mean for putting such comment utility functions

Reply:
Yes, operator_utils.h should be a better place.

Reply:
message changed locally, will commit later together with other fixes.